From 8b494b3be800316b6760d81361de2b749251164b Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Mon, 14 Jul 2025 08:09:11 +0200
Subject: [PATCH 47/N] Adjust pipe name for msys

needs to be added to "Rename dll from cygwin to msys" on the
next rebase.

Though with the installation key being part of the pipe name
I wonder if this is all needed at all, but that's another issue.
---
 winsup/cygwin/fhandler/pty.cc | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/winsup/cygwin/fhandler/pty.cc b/winsup/cygwin/fhandler/pty.cc
index d5e8864..8154fcc 100644
--- a/winsup/cygwin/fhandler/pty.cc
+++ b/winsup/cygwin/fhandler/pty.cc
@@ -1613,9 +1613,15 @@ fhandler_pty_slave::tcflush (int queue)
   if (queue == TCIFLUSH || queue == TCIOFLUSH)
     {
       char pipe[MAX_PATH];
+#ifdef __MSYS__
+      __small_sprintf (pipe,
+		       "\\\\.\\pipe\\msys-%S-pty%d-master-ctl",
+		       &cygheap->installation_key, get_minor ());
+#else
       __small_sprintf (pipe,
 		       "\\\\.\\pipe\\cygwin-%S-pty%d-master-ctl",
 		       &cygheap->installation_key, get_minor ());
+#endif
       pipe_request req = { FLUSH_INPUT, GetCurrentProcessId () };
       pipe_reply repl;
       DWORD n;
